packages/yarnpkg-core/sources/execUtils.ts

```diff
@@ -33,6 +34,15 @@ function sigintHandler() {
 // longer needed.
 let sigintRefCount = 0;
 
+const activeChildren = new Set<ChildProcess>();
+function sigtermHandler() {
+  for (const child of activeChildren) {
+    child.kill();
+  }
```

Comment:
Need to clear the set after sending the kill signals

Reply:
I think that's fine, since the processes may decide to ignore the signal until a later time.

Reply:
I don't think it matters since our process is exiting after this handler completes

Reply:
@ganemone No, we still wait for all children to terminate